test/functionalTest/cases/1875_NGSIv2_notification_formats/NGSIv2_notification_formats.test

```diff
@@ -0,0 +1,509 @@
+
+{
+    "data": [
+        {
+            "A1": {
+                "metadata": {}, 
+                "type": "none", 
+                "value": 1
+            }, 
+            "A3": {
+                "metadata": {}, 
+                "type": "none", 
+                "value": 3
+            }, 
+            "id": "E1", 
+            "type": "none"
+        }
+    ], 
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 170
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Host: localhost:REGEX(\d+)
+Accept: application/json
+X-Ngsiv2-Attrsformat: normalized
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A1": {
+                "metadata": {}, 
+                "type": "none", 
+                "value": 1
+            }, 
+            "A4": {
+                "metadata": {}, 
+                "type": "none", 
+                "value": 4
+            }, 
+            "id": "E1", 
+            "type": "none"
+        }
+    ], 
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 94
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Host: localhost:REGEX(\d+)
+Accept: application/json
+X-Ngsiv2-Attrsformat: keyvalues
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A1": 1, 
+            "A5": 5, 
+            "id": "E1", 
+            "type": "none"
+        }
+    ], 
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 60
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Host: localhost:REGEX(\d+)
+Accept: application/json
+X-Ngsiv2-Attrsformat: values
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        [
+            1, 
+            6
+        ]
+    ], 
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+} 
+=======================================
+
+
+08. Reset accumulator
+=====================
+
+
+09. Update sub2, setting attrsFormat to keyvalues
+=================================================
+HTTP/1.1 204 No Content
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+10. Update E1/A3, to see notification in sub2
+=============================================
+HTTP/1.1 204 No Content
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+11. Dump accumulator to see notification of (A1, A3) with keyvalues
+===================================================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 95
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Host: localhost:REGEX(\d+)
+Accept: application/json
+X-Ngsiv2-Attrsformat: keyvalues
```

Comment:
I think the righ one according to NGSIv2 spec is `keyValues`, not `keyvalues`. As far as I understand, HTTP header names are case-insensitive, but I think values of the headers are case-sensitive.

The same `keyValues` should be used at DB format csub level.